### PR TITLE
latencypredictor: training server mean objective + sliding window

### PR DESCRIPTION
**Pull request by kaushikmitr**
This pull request adds support for selecting between quantile and mean regression objectives in the latency predictor, refactors feature engineering to include a new binary feature, and updates model training and evaluation logic to handle both objectives. The changes provide more flexibility for model selection and improve feature representation for queueing behavior. Metrics tracking and reporting are also updated to reflect the chosen objective.

**Objective selection and metrics reporting:**
- Introduced a new `ObjectiveType` enum and added a `LATENCY_OBJECTIVE_TYPE` environment variable, allowing selection between quantile and mean regression objectives (`quantile` or `mean`). The selected objective is now stored in `Settings` and used throughout model training and evaluation. [[1]](diffhunk://#diff-76e543984bdb408957619649612903de3ff19e7b0fe332e59989752a661548f9R62-R66) [[2]](diffhunk://#diff-76e543984bdb408957619649612903de3ff19e7b0fe332e59989752a661548f9R113-R116) [[3]](diffhunk://#diff-76e543984bdb408957619649612903de3ff19e7b0fe332e59989752a661548f9L227-R235)
- Updated metric tracking to store and report either quantile metrics (quantile loss, coverage, violation rate) or mean metrics (MAE, RMSE) based on the chosen objective, including changes to logging and test evaluation. [[1]](diffhunk://#diff-76e543984bdb408957619649612903de3ff19e7b0fe332e59989752a661548f9R269-R274) [[2]](diffhunk://#diff-76e543984bdb408957619649612903de3ff19e7b0fe332e59989752a661548f9L729-R782)

**Feature engineering and model input:**
- Added a new binary feature `is_queued` to indicate whether there are queued requests, and included it in the feature columns for both TTFT and TPOT models. All model training and evaluation functions now expect this feature. [[1]](diffhunk://#diff-76e543984bdb408957619649612903de3ff19e7b0fe332e59989752a661548f9R372-R374) [[2]](diffhunk://#diff-76e543984bdb408957619649612903de3ff19e7b0fe332e59989752a661548f9R391) [[3]](diffhunk://#diff-76e543984bdb408957619649612903de3ff19e7b0fe332e59989752a661548f9R405) [[4]](diffhunk://#diff-76e543984bdb408957619649612903de3ff19e7b0fe332e59989752a661548f9L690-R724)

**Model training logic:**
- Refactored model training functions to dynamically set the regression objective and related parameters for XGBoost and LightGBM based on the selected `ObjectiveType`. The feature order is also adjusted to include or exclude queue-related features as needed. [[1]](diffhunk://#diff-76e543984bdb408957619649612903de3ff19e7b0fe332e59989752a661548f9L425-R444) [[2]](diffhunk://#diff-76e543984bdb408957619649612903de3ff19e7b0fe332e59989752a661548f9R482-R489) [[3]](diffhunk://#diff-76e543984bdb408957619649612903de3ff19e7b0fe332e59989752a661548f9L492-R517) [[4]](diffhunk://#diff-76e543984bdb408957619649612903de3ff19e7b0fe332e59989752a661548f9L502-R663)

**Evaluation logic:**
- Consolidated quantile and mean metric calculation into a single `_calculate_metrics_on_test` function, which returns the appropriate metrics based on the objective. This function replaces the previous quantile-specific metric calculation.

These changes make the latency predictor more configurable and robust, supporting both quantile and mean regression objectives and improving feature representation for queueing.

### CODE REVIEW

File: latencypredictor/training_server.py [hunk 1]
```diff
@@ -499,138 +524,143 @@ def _train_model_with_scaling(self, features: pd.DataFrame, target: pd.Series, m
...
-                objective="reg:quantileerror",    # Quantile loss (pinball loss) for quantile regression
-                quantile_alpha=self.quantile,     # Target quantile (e.g., 0.9 for p90)
-        
-                # Performance optimization:
-                tree_method='hist',          # Efficient histogram algorithm; optimal for large datasets
-                n_jobs=-1,                   # Utilize all CPU cores for parallel training
-                random_state=42,             # Ensures reproducible results
-                verbosity=1,
-                enable_categorical=True       # Enable categorical feature support   
-    )
```
Comment (BenjaminBraunDev) [inline latencypredictor/training_server.py:549]: Might be worth keeping these comments describing the dictionary entries

File: latencypredictor/training_server.py [hunk 2]
```diff
@@ -499,138 +524,143 @@ def _train_model_with_scaling(self, features: pd.DataFrame, target: pd.Series, m
...
+                    xgb_params["objective"] = "reg:quantileerror"
+                    xgb_params["quantile_alpha"] = self.quantile
+                model = xgb.XGBRegressor(**xgb_params)
                 model.fit(features, target, sample_weight=sample_weight)
                 return model
-            elif self.model_type == ModelType.LIGHTGBM:  # LightGBM with quantile regression
-                model = lgb.LGBMRegressor(
```
Comment (BenjaminBraunDev) [inline latencypredictor/training_server.py:553]: (same as above, keeping these commends helps with future changes)

File: latencypredictor/training_server.py (hunk 1 above)
Comment (kaushikmitr) [reply latencypredictor/training_server.py:549]: Done!

File: latencypredictor/training_server.py (hunk 2 above)
Comment (kaushikmitr) [reply latencypredictor/training_server.py:553]: Done!
